pages/client-libraries/go.mdx

```diff
@@ -1,130 +1,770 @@
+
+
+#### Running queries with property map
 
-func createItemFn(tx neo4j.Transaction) (interface{}, error) {
-    records, err := tx.Run(
-        "CREATE (a:Greeting) SET a.message = $message RETURN 'Node ' + id(a) + ': ' + a.message",
-        map[string]interface{}{"message": "Hello, World!"})
-    // In face of driver native errors, make sure to return them directly.
-    // Depending on the error, the driver may try to execute the function again.
-    if err != nil {
-        return nil, err
-    }
-    record, err := records.Single()
-    if err != nil {
-        return nil, err
-    }
-    // You can also retrieve values by name, with e.g. `id, found := record.Get("n.id")`
-    return &Item{
-        Message: record.Values[0].(string),
-    }, nil
+If you want to pass a map of properties to the query, you can do it like this:
+
+```go
+_, err = neo4j.ExecuteQuery(ctx, driver, "CREATE (n:Technology{name: $name, type: $type})", map[string]any{"name": "Memgraph", "type": "graph database"}, neo4j.EagerResultTransformer, neo4j.ExecuteQueryWithDatabase(""))
+if err != nil {
+	panic(err)
 }
+```
+This way your queries do not contain hard-coded values, and you can make your queries more dynamic.
 
-type Item struct {
-    Message string
+### Process the results
+
+In order to read results and serve them back to you Go application you need to handle them. Go is statically typed language and proper types need to be handled. 
+Depending on the type of the request you made, you can get different results back. Lets go over a few basic examples on how to handle different types and access properties of the returned results.
+
+#### Process the Node result
+
+By running the following query:
+```go
+result, err := neo4j.ExecuteQuery(ctx, driver, "MATCH (n:Technology{name: 'Memgraph'}) RETURN n", nil, neo4j.EagerResultTransformer, neo4j.ExecuteQueryWithDatabase("memgraph"))
+if err != nil {
+    panic(err)
 }
 ```
+When using `ExceuteQuery`, the returned results are in the `EagerResults` format, which means you have access to the `Records` field. Records field contains all the records returned by the query. To process the results, you can iterate over the records and access the fields you need. 
+Here is the example: 
+
+```go
+for _, node := range result.Records {
+    fmt.Println(node.AsMap()["n"].(neo4j.Node))
+}
+```
+```text
+{65 65 [Technology] map[createdAt:{0 63828259200 <nil>} description:Fastest graph DB in the world! id:1 name:Memgraph]}
+```
+
+Lets break down the result. So for each returned record, it is getting converted to a map, now we can access the `n` field, which is a `Node` returned from a query. The returned record and all its properties are of type `any`. This means that you have to cast them to the relevant Go type if you want to use methods/features defined on such types. 
+To access individual properties of the `Node` you have following options: 
```

Comment:
This sounds ok to me, You need to say Go what type it will recive, once the Type is there you can call some function. For example if type is Path you can call Get source node etc.